### PR TITLE
Remove hardfork activation time overrides

### PR DESCRIPTION
**Pull request by geoknee**
This is to be coordinated with https://github.com/ethereum-optimism/superchain-registry/pull/101. See that PR and linked issue for full context.

### CODE REVIEW

**Comment by geoknee** [comment]
This existing test would fail if the migration of the overrides (from here to `superchain-registry`) was not performed correctly: 
https://github.com/ethereum-optimism/optimism/blob/97b107b62a35b6e6188af4a800952f2932823a72/op-node/chaincfg/chains_test.go#L13-L22

File: op-node/rollup/superchain.go [hunk 1]
```diff
@@ -74,14 +74,6 @@ func LoadOPStackRollupConfig(chainID uint64) (*Config, error) {
 		regolithTime = 1692156862
 	}
 
-	deltaTime := superChain.Config.DeltaTime
-	// OP Labs Sepolia devnet 0 activated delta at genesis, slightly earlier than
-	// Base Sepolia devnet 0 on the same superchain.
-	switch chainID {
-	case labsSepoliaDevnet0:
-		deltaTime = new(uint64)
-	}
```
Comment (sebastianst) [inline op-node/rollup/superchain.go:83]: We should add a test to `op-node/chaincfg/chains_test.go` for this network. It should fail right now.

File: go.mod [hunk 2]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/crate-crypto/go-kzg-4844 v0.7.0
 	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.2.0
 	github.com/ethereum-optimism/go-ethereum-hdwallet v0.1.3
-	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20240222155908-ab073f6aa74f
+	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20240227110823-11c73f54ad9b
```
Comment (sebastianst) [inline go.mod]: need to update this once we put the op-sepolia-devnet-0 delta override into the superchain-registry pr.

File: op-node/rollup/superchain.go (hunk 1 above)
Comment (geoknee) [reply op-node/rollup/superchain.go:83]: See [988e48e](https://github.com/ethereum-optimism/optimism/pull/9642/commits/988e48e5aa4bab93209a3f9022d5ffc6f44e0b75) and [bc10117](https://github.com/ethereum-optimism/optimism/pull/9642/commits/bc10117b06db0fbbd460474880c51f93ffa354a3).

Notes: 
* To construct the expected config for this chain, I copied everything over from `superchain-registry`
* This is the only devnet chain in this test suite

File: op-node/chaincfg/chains_test.go [hunk 3]
```diff
@@ -136,6 +137,41 @@ var sepoliaCfg = rollup.Config{
...
+	L2ChainID:               big.NewInt(11155421),
+	BatchInboxAddress:       common.HexToAddress("0xff00000000000000000000000000000011155421"),
+	DepositContractAddress:  common.HexToAddress("0x76114bd29dFcC7a9892240D317E6c7C2A281Ffc6"),
+	L1SystemConfigAddress:   common.HexToAddress("0xa6b72407e2dc9EBF84b839B69A24C88929cf20F7"),
+	RegolithTime:            u64Ptr(0),
+	CanyonTime:              u64Ptr(1698436800),
+	DeltaTime:               u64Ptr(0),
+	EcotoneTime:             u64Ptr(1706634000),
+	ProtocolVersionsAddress: common.HexToAddress("0x252CbE9517F731C618961D890D534183822dcC8d"),
+}
```
[the remark this answers is not in the document]
Comment (geoknee) [reply op-node/chaincfg/chains_test.go]: No changes are necessary, the TODO on L140 can be removed.

File: op-node/rollup/superchain.go (hunk 1 above)
Comment (geoknee) [reply op-node/rollup/superchain.go:83]: * it fails as expected at commit [bc10117](https://github.com/ethereum-optimism/optimism/pull/9642/commits/bc10117b06db0fbbd460474880c51f93ffa354a3)

Comment (geoknee) [reply op-node/rollup/superchain.go:83]: * And passes after commit [2bb74bc](https://github.com/ethereum-optimism/optimism/pull/9642/commits/2bb74bc64f443adb536e6edb37bef81b4f078bb9)

File: go.mod (hunk 2 above)
Comment (geoknee) [reply go.mod]: [2bb74bc](https://github.com/ethereum-optimism/optimism/pull/9642/commits/2bb74bc64f443adb536e6edb37bef81b4f078bb9)

File: op-node/chaincfg/chains_test.go [hunk 4]
```diff
@@ -136,6 +137,40 @@ var sepoliaCfg = rollup.Config{
...
+	MaxSequencerDrift:       600,
+	SeqWindowSize:           3600,
+	ChannelTimeout:          300,
+	L1ChainID:               big.NewInt(11155111),
+	L2ChainID:               big.NewInt(11155421),
+	BatchInboxAddress:       common.HexToAddress("0xff00000000000000000000000000000011155421"),
+	DepositContractAddress:  common.HexToAddress("0x76114bd29dFcC7a9892240D317E6c7C2A281Ffc6"),
+	L1SystemConfigAddress:   common.HexToAddress("0xa6b72407e2dc9EBF84b839B69A24C88929cf20F7"),
+	RegolithTime:            u64Ptr(0),
+	CanyonTime:              u64Ptr(1698436800),
```
Comment (geoknee) [inline op-node/chaincfg/chains_test.go]: I want to double check this is correct. I suspect that maybe it should be 0.

Comment (trianglesphere) [reply op-node/chaincfg/chains_test.go]: Sepolia devnet had some weird stuff with the fork timestamps cc @sebastianst

Comment (geoknee) [reply op-node/chaincfg/chains_test.go]: Thanks @trianglesphere, we are tracking that here https://github.com/ethereum-optimism/client-pod/issues/624

Comment (geoknee) [reply op-node/chaincfg/chains_test.go]: [ab98d79](https://github.com/ethereum-optimism/optimism/pull/9642/commits/ab98d79d24753fb8915d3a4cc013b267b11c02ed)

**Review by sebastianst — APPROVED** [review]

**Review by ajsutton — APPROVED** [review]

**Pull request merged**